dropshot/tests/test_config.rs

```diff
@@ -140,13 +233,124 @@ async fn test_config_bind_address() {
+        // Configure TLS to trust the self-signed cert
+        let mut root_store = rustls::RootCertStore { roots: vec![] };
+        root_store.add(&certs[certs.len() - 1]).expect("adding root cert");
+
+        let tls_config = rustls::ClientConfig::builder()
+            .with_safe_defaults()
+            .with_root_certificates(root_store)
+            .with_no_client_auth();
+        let https_connector = hyper_rustls::HttpsConnectorBuilder::new()
+            .with_tls_config(tls_config)
+            .https_only()
+            .enable_http1()
+            .build();
+        hyper::Client::builder().build(https_connector)
+    };
+
+    let client = make_client();
+    let hostname = "localhost";
+    let bind_ip_str = "127.0.0.1";
+    /* This must be different than the bind_port used in the http test. */
+    let bind_port = 12217;
+
+    /*
+     * This helper constructs a GET HTTP request to
+     * http://$bind_ip_str:$port/, where $port is the argument to the
+     * closure.
+     */
+    let cons_request = |port: u16| {
+        let uri = hyper::Uri::builder()
+            .scheme("https")
+            .authority(format!("{}:{}", hostname, port).as_str())
+            .path_and_query("/")
+            .build()
+            .unwrap();
+        hyper::Request::builder()
+            .method(http::method::Method::GET)
+            .uri(&uri)
+            .body(hyper::Body::empty())
+            .unwrap()
+    };
+
+    /*
+     * Make sure there is not currently a server running on our expected
+     * port so that when we subsequently create a server and run it we know
+     * we're getting the one we configured.
+     */
+    let error = client.request(cons_request(bind_port)).await.unwrap_err();
+    assert!(error.is_connect());
+
+    /*
+     * Now start a server with our configuration and make the request again.
+     * This should succeed in terms of making the request.  (The request
+     * itself might fail with a 400-level or 500-level response code -- we
+     * don't want to depend on too much from the ApiServer here -- but we
+     * should have successfully made the request.)
+     */
+    let tls = Some(TlsConfig {
+        cert_file: cert_file.path().to_str().unwrap(),
+        key_file: key_file.path().to_str().unwrap(),
+    });
+    let config = make_config(bind_ip_str, bind_port, tls);
+    let server = make_server(&config, &log).start();
+    client.request(cons_request(bind_port)).await.unwrap();
+    server.close().await.unwrap();
+
+    /*
+     * Make another request to make sure it fails now that we've shut down
+     * the server.  We need a new client to make sure our client-side connection
+     * starts from a clean slate.  (Otherwise, a race during shutdown could
+     * cause us to successfully send a request packet, only to have the TCP
+     * stack return with ECONNRESET, which gets in the way of what we're trying
+     * to test here.)
+     */
+    let client = make_client();
```

Comment:
It seems like a bunch of this function is duplicated with the earlier one.  Is it worth trying to share code here?

Reply:
I think so, in case we want to expand any test cases. Changed it